addons/mrp/models/mrp_workorder.py

```diff
@@ -520,6 +520,48 @@ def _start_nextworkorder(self):
                  float_compare(self.operation_id.batch_size, self.qty_produced, precision_rounding=rounding) <= 0)):
             self.next_work_order_id.state = 'ready'
 
+    def _get_unavailability_mapping(self, workcenter_ids, start_datetime, end_datetime):
+        workcenters = self.env['mrp.workcenter'].browse(workcenter_ids)
+        return workcenters.mapped('resource_id')._get_unavailable_intervals(start_datetime, end_datetime)
+
+    @api.model
+    def gantt_unavailability(self, start_date, end_date, scale, group_bys=None, rows=None):
```

Comment:
it would be nice to not do the first traverse_inplace, maybe you can infer the workcenters from the group_bys args? i don't know

Reply:
Because the data is in the resId object in the tree corresponding of the groupBy node, it is not possible :/